app/services/make_offer.rb

```diff
@@ -16,10 +16,7 @@ def save!
                                     application_choice: application_choice,
                                     course_option: course_option,
                                     offer_conditions: conditions)
-    make_an_offer.save
 
-    if make_an_offer.errors[:base].include?(MakeAnOffer::STATE_TRANSITION_ERROR)
-      ApplicationStateChange.new(application_choice).make_offer!
-    end
+    raise 'Unable to complete save on make_an_offer' unless make_an_offer.save
```

Comment:
As the only error added to MakeAnOffer is the state transition error, and this behavior (raising state transition errors) is replicated across other services I don't think this change is required as it enables a better understanding of why this exception occured..

Reply:
You are right no other error should normally occur, based on the current code, but I think what we have here is safer. Checking for the inclusion of a specific error and raising an exception only if found feels too specific, as we essentially ignore the return value of `save`.

Reply:
I've had a think about this and I think we can support both cases:
1) if we have the ability to surface a granular error we should do so
2) We should also guard against unexpected errors that might occur in the wrapped class past the validations.

Maybe something similar to this will work, where we surface a transition error if it has occurred, but also raise the other error otherwise:

```
unless make_an_offer.save
  if make_an_offer.errors[:base].include?(MakeAnOffer::STATE_TRANSITION_ERROR)
    ApplicationStateChange.new(application_choice).make_offer!
  else
    raise 'Unable to complete save on make_an_offer'
  end
end
```